### ISSUE TITLE
no auto-generation of methods from associations

### ISSUE THREAD

**Issue by musicformellons** [feathersjs/feathers#741]
I am trying to setup a m:n association using feathersjs with sequelize. I am following this ['instruction'][1], but have to adjust to updates feathersjs has gone throught in the meantime.

My hook to associate an already created user with an already created item:

    module.exports = function (options = {}) { 
      return function (hook) {
        hook.app.service('users')
          .get(hook.data[0].user_id)
          .then(users => {
            hook.app.service('items')
              .find(hook.data[0].item_id)
              .then(benches => items.addMyusers(users));
          });
    
        return Promise.resolve(hook);
      };
    };

I get:

    error: Unhandled Rejection at: Promise  Promise {
      <rejected> TypeError: Cannot read property 'addMyusers' of undefined

Under my user model definition I set the asssocation with:

      users.associate = function (models) {
        users.belongsToMany(models.items, {
          through: {
            model: 'favorites-items',
            unique: false
          },
          foreignKey: 'item_id',
          as: 'myitems',
          constraints: false
        });
      };

under item model definition:

      items.associate = function (models) { 
        items.belongsToMany(models.users, {
          through: {
            model: 'favorites-items',
            unique: false
          },
          as: 'myusers',
          foreignKey: 'user_id',
          constraints: false
        });
      };

Why does addMyusers not exist? Methods like these should be auto-generated when making the association?

As stated in the [sequelize docs][2]: 

> This will add methods getUsers, setUsers, addUser,addUsers to Project,
> and getProjects, setProjects, addProject, and addProjects to User.

  [1]: https://www.youtube.com/watch?v=TUEhRlKxrOQ
  [2]: http://docs.sequelizejs.com/manual/tutorial/associations.html

**Comment by daffl** [comment]
Try setting `raw: false` as [documented in the feathers-sequelize documentation](https://docs.feathersjs.com/api/databases/sequelize.html#a-quick-note-about-raw-queries).

**Comment by musicformellons** [comment]
I'm struggling... I tried now:

My `associateExistingUserToExistingPoint.js` hook to associate an already created user with an already created item (point):

```
module.exports = function (options = {}) { 
  return function (hook) {
    if (!hook.params.sequelize) hook.params.sequelize = {};
    Object.assign(hook.params.sequelize, {raw: false});

    hook.app.service('users')
      .get(hook.data[0].user_id)
      .then(user => {
        hook.app.service('points')
          .get(Number(hook.data[0].point_id))
          .then(point => point.addMyusers(user));
      });

    return Promise.resolve(hook);
  };
};
```
I am also using `addRelationship.js` hooks for both user and point services. For points:
```
module.exports = function (options = {}) { 
  return function (hook) {
    const AssociatedModel = hook.app.service('users').Model;
    hook.params.sequelize = {
      raw: false,
      include: [{
        model: AssociatedModel,
        as: 'myusers',
        attributes: ['id']
      }]
    };

    return Promise.resolve(hook);
  };
};
```

For users:
```
module.exports = function (options = {}) { 
  return function (hook) {
    const AssociatedModel = hook.app.service('points').Model;
    hook.params.sequelize = {
      raw: false,
      include: [{
        model: AssociatedModel,
        as: 'mypoints',
        attributes: ['point_id']
      }]
    };

    return Promise.resolve(hook);
  };
};

```
The output I get when adding creating favorites_points (linking user_id to point_id as a m:n join table) using this on the client:
```
      api.service('/favorites_points').create(
        [{
          'user_id': this.userId,
          'point_id': pointId
        }]
      )
```
, first entry:
```
{ data: [ { user_id: 1, point_id: '3' } ],
  params: 
   { query: {},
     provider: 'socketio',
     payload: { userId: 1 },
     user: 
      users {
        dataValues: [Object],
        _previousDataValues: [Object],
        _changed: {},
        _modelOptions: [Object],
        _options: [Object],
        __eagerlyLoadedAssociations: [],
        isNewRecord: false,
        mypoints: [] },
     accessToken: 'mytoken',
     headers: 
      { Authorization: 'mytoken' },
     authenticated: true,
     sequelize: { raw: false } },
  callback: [Function: callbackWrapper],
  method: 'create',
  type: 'before',
  app: 
   { [EventEmitter: app]
     domain: undefined,
     _events: { mount: [Array] },
     _maxListeners: undefined,
     setMaxListeners: [Function: setMaxListeners],

[ETCETERA...]

     filter: [Function: filter],
     _super: undefined,
     _eventsCount: 4,
     _socketSetup: true },
  path: 'favorites_points' }
  
info: after: favorites_points - Method: create
info: after: users - Method: get
info: error: points - Method: get: No record found for id '3'
error:  NotFound: No record found for id '3'
    at new NotFound (/home/usr/api_server/node_modules/feathers-errors/lib/index.js:121:17)
    at /home/usr/api_server/node_modules/feathers-sequelize/lib/index.js:175:15
    at tryCatcher (/home/usr/api_server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/usr/api_server/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/usr/api_server/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/usr/api_server/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
error: Unhandled Rejection at: Promise  Promise {
  <rejected> { NotFound: No record found for id '3'
    at new NotFound (/home/usr/api_server/node_modules/feathers-errors/lib/index.js:121:17)
    at /home/usr/api_server/node_modules/feathers-sequelize/lib/index.js:175:15
    at tryCatcher (/home/usr/api_server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/usr/api_server/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/usr/api_server/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/usr/api_server/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
  type: 'FeathersError',
  name: 'NotFound',
  message: 'No record found for id \'3\'',
  code: 404,
  className: 'not-found',
  data: 
   { name: 'SequelizeDatabaseError',
     parent: [Object],
     original: [Object],
     sql: 'SELECT "points"."point_id", "points"."url", "points"."city", "points"."image", "points"."wkb_geometry", "myusers"."id" AS "myusers.id", "myusers->favorites_points"."user_id" AS "myusers.favorites_points.user_id", "myusers->favorites_points"."point_id" AS "myusers.favorites_points.point_id", "myusers->favorites_points"."createdAt" AS "myusers.favorites_points.createdAt", "myusers->favorites_points"."updatedAt" AS "myusers.favorites_points.updatedAt" FROM "points" AS "points" LEFT OUTER JOIN ( "favorites_points" AS "myusers->favorites_points" INNER JOIN "users" AS "myusers" ON "myusers"."id" = "myusers->favorites_points"."point_id") ON "points"."point_id" = "myusers->favorites_points"."user_id" WHERE "points"."id" = 3;' },
  errors: {} } } NotFound: No record found for id '3'
    at new NotFound (/home/usr/api_server/node_modules/feathers-errors/lib/index.js:121:17)
    at /home/usr/api_server/node_modules/feathers-sequelize/lib/index.js:175:15
    at tryCatcher (/home/usr/api_server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/usr/api_server/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/usr/api_server/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/usr/api_server/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
```
    

Second entry:

```
 {data: [ { user_id: 1, point_id: '1' } ],
  params: 
   { query: {},
     provider: 'socketio',
     payload: { userId: 1 },
     user: 
      users {
        dataValues: [Object],
        _previousDataValues: [Object],
        _changed: {},
        _modelOptions: [Object],
        _options: [Object],
        __eagerlyLoadedAssociations: [],
        isNewRecord: false,
        mypoints: [] },
     accessToken: 'eyJhbGciOiJIUzI1NiIsInR5cCI6ImFjY2VzcyIsInR5cGUiOiJhY2Nlc3MifQ.eyJ1c2VySWQiOjEsImlhdCI6MTUxMjI0NjgyMiwiZXhwIjoxNTEyMzMzMjIyLCJhdWQiOiJodHRwczovL3BxYnEueHl6IiwiaXNzIjoiZmVhdGhlcnMiLCJzdWIiOiJhbm9ueW1vdXMiLCJqdGkiOiI2OWM2OWM0YS1mYzI4LTRjZWYtODgyZC00YmNlMWYxZGVlM2YifQ.I-iTLVXFrFCIFw-EQdi0uNDCOx-pZ-pcXiq0dBD82BQ',
     headers: 
      { Authorization: 'eyJhbGciOiJIUzI1NiIsInR5cCI6ImFjY2VzcyIsInR5cGUiOiJhY2Nlc3MifQ.eyJ1c2VySWQiOjEsImlhdCI6MTUxMjI0NjgyMiwiZXhwIjoxNTEyMzMzMjIyLCJhdWQiOiJodHRwczovL3BxYnEueHl6IiwiaXNzIjoiZmVhdGhlcnMiLCJzdWIiOiJhbm9ueW1vdXMiLCJqdGkiOiI2OWM2OWM0YS1mYzI4LTRjZWYtODgyZC00YmNlMWYxZGVlM2YifQ.I-iTLVXFrFCIFw-EQdi0uNDCOx-pZ-pcXiq0dBD82BQ' },
     authenticated: true,
     sequelize: { raw: false } },
  callback: [Function: callbackWrapper],
  method: 'create',
  type: 'before',
  app: 
   { [EventEmitter: app]
     domain: undefined,
     _events: { mount: [Array] },
     
[ETCETERA...]

     _serviceEvents: [ 'created', 'updated', 'removed', 'patched' ],
     setup: [Function],
     _eventFilters: { all: [Array] },
     filter: [Function: filter],
     _super: undefined,
     _eventsCount: 4,
     _socketSetup: true },
  path: 'favorites_points' }
info: after: favorites_points - Method: create
info: after: users - Method: get
info: error: points - Method: get: No record found for id '1'
error:  NotFound: No record found for id '1'
    at new NotFound (/home/usr/api_server/node_modules/feathers-errors/lib/index.js:121:17)
    at /home/usr/api_server/node_modules/feathers-sequelize/lib/index.js:175:15
    at tryCatcher (/home/usr/api_server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/usr/api_server/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/usr/api_server/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/usr/api_server/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
error: Unhandled Rejection at: Promise  Promise {
  <rejected> { NotFound: No record found for id '1'
    at new NotFound (/home/usr/api_server/node_modules/feathers-errors/lib/index.js:121:17)
    at /home/usr/api_server/node_modules/feathers-sequelize/lib/index.js:175:15
    at tryCatcher (/home/usr/api_server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/usr/api_server/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/usr/api_server/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/usr/api_server/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
  type: 'FeathersError',
  name: 'NotFound',
  message: 'No record found for id \'1\'',
  code: 404,
  className: 'not-found',
  data: 
   { name: 'SequelizeDatabaseError',
     parent: [Object],
     original: [Object],
     sql: 'SELECT "points"."point_id", "points"."url", "points"."city", "points"."image", "points"."wkb_geometry", "myusers"."id" AS "myusers.id", "myusers->favorites_points"."user_id" AS "myusers.favorites_points.user_id", "myusers->favorites_points"."point_id" AS "myusers.favorites_points.point_id", "myusers->favorites_points"."createdAt" AS "myusers.favorites_points.createdAt", "myusers->favorites_points"."updatedAt" AS "myusers.favorites_points.updatedAt" FROM "points" AS "points" LEFT OUTER JOIN ( "favorites_points" AS "myusers->favorites_points" INNER JOIN "users" AS "myusers" ON "myusers"."id" = "myusers->favorites_points"."point_id") ON "points"."point_id" = "myusers->favorites_points"."user_id" WHERE "points"."id" = 1;' },
  errors: {} } } NotFound: No record found for id '1'
    at new NotFound (/home/usr/api_server/node_modules/feathers-errors/lib/index.js:121:17)
    at /home/usr/api_server/node_modules/feathers-sequelize/lib/index.js:175:15
    at tryCatcher (/home/usr/api_server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/usr/api_server/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/usr/api_server/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/usr/api_server/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/usr/api_server/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
info: after: users - Method: find

```

When I run this from client then:
```
      api.service('users').find({
        user_id: this.userId,
        sequelize:
          {
            include: [{
              model: 'mypoints'
            }]
          }
      })
```

I get this output in the browser:

![screenshot from 2017-12-02 22-38-29](https://user-images.githubusercontent.com/10334123/33519959-ad6c6af4-d7b1-11e7-8c75-d9682105870c.png)

So I get my user but only receive one favorite instead of the two (or more...) added. And actually it only shows the point with id 1. When I do not make that one a favorite it shows none...

Hope the above makes sense... Would be great if you could get me floating.

**Comment by musicformellons** [comment]
I am using a custom join table with modelname `favoritesPoints` and tablename `favorites_points`.
In the [sequelize API reference docs](http://docs.sequelizejs.com/class/lib/model.js~Model.html) it says:

> options.through | Model \| string \| object |   | The name of the table that is used to join source and target in n:m associations. Can also be a sequelize model if you want to define the junction table yourself and add extra attributes to it.
> options.through.model | Model | optional | The model used to join both sides of the N:M association.

So I am not sure if I can reference by tablename (and not the modelname) of my join-table as I am currently doing. Just using `through: 'tablename'` is fine, but when giving an extra options object to through maybe it needs the actual model?! But then I need to import it first, so how do I do that in the current default feathersjs setup (i tried but did not work...)?

```
    users.belongsToMany(models.points, {
      through: {
        model: 'favorites_points',
        unique: false
      },
```

**Comment by musicformellons** [comment]
Yeah, just got confirmed from sequelize slack channel that I need to refer with model to actual model (when using a string it will create the join table instead of using the custom one). So how do i import my favoritesPoints model into my users.model.js in the current feathersjs cli structure?

**Comment by musicformellons** [comment]
when I do (in both users and points model):
`const favoritesPoints = require('./favorites_points.model');`
with:
```
    users.belongsToMany(models.points, {
      through: {
        model: favoritesPoints,
        unique: false
      },
```
I get following error:
```
/home/usr/api_server/node_modules/sequelize/lib/associations/belongs-to-many.js:266
    if (!this.through.model.rawAttributes[this.foreignKey]) {
                                         ^

TypeError: Cannot read property 'user_id' of undefined
    at BelongsToMany.injectAttributes (/home/usr/api_server/node_modules/sequelize/lib/associations/belongs-to-many.js:266:42)
    at Function.belongsToMany (/home/usr/api_server/node_modules/sequelize/lib/associations/mixin.js:53:17)
    at Function.users.associate (/home/usr/api_server/src/models/users.model.js:52:11)
    at Object.values.filter.forEach.model (/home/usr/api_server/src/sequelize.js:36:22)
    at Array.forEach (<anonymous>)
    at Function.app.setup [as _super] (/home/usr/api_server/src/sequelize.js:35:8)
    at Function.setup (/home/usr/api_server/node_modules/feathers-socketio/lib/index.js:66:28)
    at Function.<anonymous> (/home/usr/api_server/node_modules/uberproto/lib/proto.js:30:17)
    at Function.app.setup (/home/usr/api_server/node_modules/feathers-authentication/lib/index.js:92:27)
    at Function.app.setup (/home/usr/api_server/node_modules/feathers-authentication-jwt/lib/index.js:83:27)
    at Function.app.setup (/home/usr/api_server/node_modules/feathers-authentication-local/lib/index.js:69:27)
    at Function.app.setup (/home/usr/api_server/node_modules/feathers-authentication-oauth2/lib/index.js:101:27)
    at Function.app.setup (/home/usr/api_server/node_modules/feathers-authentication-oauth2/lib/index.js:101:27)
    at Function.listen (/home/usr/api_server/node_modules/feathers/lib/application.js:157:10)
    at Function.listen (/home/usr/api_server/node_modules/uberproto/lib/proto.js:30:17)
    at Object.<anonymous> (/home/usr/api_server/src/index.js:5:20)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Function.Module.runMain (module.js:682:10)
npm ERR! code ELIFECYCLE

```
Which I think relates to a 'circular reference'. How do I solve that?

My user model:

```
const Sequelize = require('sequelize');
const DataTypes = Sequelize.DataTypes;
const favoritesPoints = require('./favorites_points.model');

module.exports = function (app) {
  const sequelizeClient = app.get('sequelizeClient');
  const users = sequelizeClient.define('users', {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true
    },
    email: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true
    },
    password: {
      type: DataTypes.STRING,
      allowNull: false
    },
    googleId: {
      type: DataTypes.STRING
    },
    facebookId: {
      type: DataTypes.STRING
    },
    username: {
      type: DataTypes.STRING,
      allowNull: true,
      unique: true
    }
  }, {
    hooks: {
      beforeCount(options) {
        options.raw = true;
      }
    }
);

  users.associate = function (models) {
    users.belongsToMany(models.points, {
      through: {
        // model: models.favoritesPoints,
        model: 'favorites_points',
        unique: false
      },
      foreignKey: {
        name: 'user_id',
        allowNull: false
      },
      as: 'mypoints',
      constraints: false
    });
  };

  return users;
};

```

My points model:

```
const Sequelize = require('sequelize');
const DataTypes = Sequelize.DataTypes;
const favoritesPoints = require('./favorites_points.model');

module.exports = function (app) {
  const sequelizeClient = app.get('sequelizeClient');
  const points = sequelizeClient.define('points', {
    point_id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      field: 'point_id',
      primaryKey: true
    },
    url: {
      type: DataTypes.TEXT,
      validate: {isUrl: true},
      field: 'url'
    },
    tag: {
      type: DataTypes.TEXT,
      field: 'tag'
    }
  }, {
    hooks: {
      beforeCount(options) {
        options.raw = true;
      }
    }
  });

  points.associate = function (models) {
    points.belongsToMany(models.users, {
      through: {
        model: favoritesPoints,
        unique: false
      },
      as: 'myusers',
      foreignKey: {
        name: 'point_id',
        allowNull: false
      },
      constraints: false
    });
  };

  return points;
};

```
Custom join table model:

```
const Sequelize = require('sequelize');
const DataTypes = Sequelize.DataTypes;

module.exports = function (app) {
  const sequelizeClient = app.get('sequelizeClient');
  const favoritesPoints = sequelizeClient.define('favorites_points', {
    user_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      unique: 'compositeIndex'
    },
    point_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      unique: 'compositeIndex'
    }
  }, {
    hooks: {
      beforeCount(options) {
        options.raw = true;
      }
    }
  });

  favoritesPoints.associate = function (models) { 
    // Define associations here
    // See http://docs.sequelizejs.com/en/latest/docs/associations/
  };

  return favoritesPoints;
};

```

**Comment by musicformellons** [comment]
I made a copy of my project to be able to 'freely' experiment. Things are starting to work now... Couple of things for sure caused problems:
- The `raw: false` is necessary for sequelize model use.
- I synced per table (instead of `sequelize.sync`) to have more control when I would like to keep entries of a certain table, however this way no 'through' tables are created and no 'magic formulas'. I found a stackoverflow issue, so I am not alone on this one.
- I have a hook with a 'get' in it but had not set the proper id of my service, see[ this one](https://github.com/feathersjs-ecosystem/feathers-knex/issues/101). 

Furthermore I kicked authorisation hooks out of window while 'experimenting'. Anyway, closing my messy issue now.
